cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -321,6 +331,14 @@ func (c *CStorVolumeReplicaController) cVRAddEventHandler(
 
 	// IsEmptyStatus is to check if initial status of cVR object is empty.
 	if IsEmptyStatus(cVR) || IsInitStatus(cVR) || IsRecreateStatus(cVR) {
+		// We should generate replicaID for new volume only
+		if IsEmptyStatus(cVR) || IsInitStatus(cVR) {
+			if err := volumereplica.GenerateReplicaID(cVR); err != nil {
```

Comment:
After generating replicaId we need to update in etcd on the success we need to create zfs data set

Reply:
yes. it is updating.